File: 029-数组中出现次数超过一半的数字/029-MoreThanHalfNum_Solution.py
# -*- coding:utf-8 -*-
class Solution:
    def MoreThanHalfNum_Solution(self, numbers):
        # write code here
        dic=dict()
        for i in numbers:
            if i in dic:
                dic[i]+=1
            else:
                dic[i]=1
        for key,value in dic.items():
            if value>(len(numbers)//2):
                return key
        return 0

File: 029-数组中出现次数超过一半的数字/029-MoreThanHalfNum_Solution.cpp
class Solution {
public:
    int MoreThanHalfNum_Solution(vector<int> numbers) {
        map<int,int> hmap;
        for(auto i :numbers){
            if(hmap.find(i)!=hmap.end()){
                hmap[i]+=1; 
            }
            else{
                hmap[i]=1;
            }
        }
        for(auto it=hmap.begin();it!=hmap.end();it++){
            if((it->second) > (numbers.size()/2)){
                return it->first;
            }
        }
        return 0;
    }
};
